Solutions/Corvus.Extensions.Newtonsoft.Json/Corvus/Extensions/Json/Internal/DateTimeOffsetConverter.cs

```diff
@@ -25,7 +25,7 @@ public override bool CanConvert(Type objectType)
         }
 
         /// <inheritdoc/>
-        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        public override object? ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
```

Comment:
I'm a bit uncertain about this one, but given that `DateTimeOffset` is a struct, is it valid to return a `null` from this method? I can see this converter is intended to convert both `DateTimeOffset` and `DateTimeOffset?`, but in the case where it's converting to a non-nullable field, what's the implication of returning `null` rather than `default(DateTimeOffset)`? This might be one for @idg10 to comment on!

Reply:
Is this a question about the return type being `object?` vs `object`?

If so, I think we should take our lead from where we know Json.NET is going. They added nullable references support in V12.0.3, and here's how the base method is now defined:

```csharp
public abstract object? ReadJson(JsonReader reader, Type objectType, object? existingValue, JsonSerializer serializer);
```

So we should follow suit. Otherwise, whenever we eventually modify this project to refer to a version of Json.NET that includes the nullability annotations, we'll get an error. (The main reason we're not already doing this is that we think a few things don't currently work with Json.NET v12. E.g., the cosmos db client libraries!)